components/dashboard/src/teams/TeamSettings.tsx

```diff
@@ -110,37 +116,47 @@ export default function TeamSettings() {
     return (
         <>
             <OrgSettingsPage>
-                <Heading2>Organization Name</Heading2>
-                <Subheading className="max-w-2xl">
-                    This is your organization's visible name within Gitpod. For example, the name of your company.
-                </Subheading>
-                {errorMessage && (
+                <Heading2>Organization Details</Heading2>
+                <Subheading className="max-w-2xl">Details of your organization within Gitpod.</Subheading>
+
+                {updateOrg.isError && (
                     <Alert type="error" closable={true} className="mb-2 max-w-xl rounded-md">
-                        {errorMessage}
+                        <span>Failed to update organization information: </span>
+                        <span>{updateOrg.error.message || "unknown error"}</span>
                     </Alert>
                 )}
                 {updated && (
                     <Alert type="message" closable={true} className="mb-2 max-w-xl rounded-md">
                         Organization name has been updated.
                     </Alert>
                 )}
-                <div className="flex flex-col lg:flex-row">
-                    <div>
-                        <div className="mt-4 mb-3">
-                            <h4>Name</h4>
-                            <input type="text" value={teamName} onChange={onNameChange} />
-                        </div>
-                    </div>
-                </div>
-                <div className="flex flex-row">
-                    <button
-                        className="primary"
-                        disabled={org?.name === teamName || !!errorMessage}
-                        onClick={updateTeamInformation}
+                <form onSubmit={updateTeamInformation}>
+                    <TextInputField
+                        label="Name"
+                        hint="The name of your company or organization"
+                        value={teamName}
+                        error={teamNameError.message}
+                        onChange={setTeamName}
+                        onBlur={teamNameError.onBlur}
+                    />
+
+                    <TextInputField
+                        label="Slug"
```

Comment:
**praise**: Thanks @AlexTugarev for using the new form components! Thanks also @selfcontained for initially adding[[1](https://github.com/gitpod-io/gitpod/blob/0a5235987358b7a935f572d4a1ff82f7207291ed/components/dashboard/src/components/forms/InputField.tsx#LL28C21-L28C21)] the [`htmlFor`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLLabelElement/htmlFor) element.